### PR TITLE
Update Kotlin compiler to 1.8.20

### PR DESCRIPTION
**Pull request by vmishenev**

### CODE REVIEW

File: subprojects/analysis-kotlin-descriptors/compiler/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/compiler/configuration/AnalysisEnvironment.kt [hunk 1]
```diff
@@ -528,7 +530,14 @@ class AnalysisEnvironment(
     }
 
     // Set up JDK classpath roots explicitly because of https://github.com/JetBrains/kotlin/commit/f89765eb33dd95c8de33a919cca83651b326b246
-    internal fun configureJdkClasspathRoots() = configuration.configureJdkClasspathRoots()
```
Comment (IgnatBeresnev) [inline subprojects/analysis-kotlin-descriptors/compiler/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/compiler/configuration/AnalysisEnvironment.kt:531]: Was this function removed from the compiler with no replacement, or what was the motivation to copy paste the body?

File: subprojects/analysis-kotlin-descriptors/compiler/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/compiler/configuration/resolve/DokkaJsResolverForModuleFactory.kt [hunk 2]
```diff
@@ -42,6 +42,10 @@ internal class DokkaJsResolverForModuleFactory(
         )
     }
 
+    @Deprecated(
```
Comment (IgnatBeresnev) [inline subprojects/analysis-kotlin-descriptors/compiler/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/compiler/configuration/resolve/DokkaJsResolverForModuleFactory.kt:45]: These modules are 100% internal, I think there's no real benefit in deprecating things, and they can just be removed. If you delete it, make sure to also delete the other deprecated `createResolverForModule`, for some reason there are two..

**Review by IgnatBeresnev — APPROVED** [review]

File: subprojects/analysis-kotlin-descriptors/compiler/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/compiler/configuration/AnalysisEnvironment.kt (hunk 1 above)
Comment (vmishenev) [reply subprojects/analysis-kotlin-descriptors/compiler/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/compiler/configuration/AnalysisEnvironment.kt:531]: What do you mean? Whose body?
It copies nothing.
`configuration.configureJdkClasspathRoots()` still exists, it is used here too.

Comment (IgnatBeresnev) [reply subprojects/analysis-kotlin-descriptors/compiler/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/compiler/configuration/AnalysisEnvironment.kt:531]: I don't understand where the rest of the new code in this function came from, and why it's needed :) As far as I can see, it [already checks](https://github.com/JetBrains/kotlin/blob/1.9.0/compiler/cli/cli-base/src/org/jetbrains/kotlin/cli/jvm/config/JvmContentRoots.kt#L94) for `java.home`

Comment (vmishenev) [reply subprojects/analysis-kotlin-descriptors/compiler/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/compiler/configuration/AnalysisEnvironment.kt:531]: > I don't understand where the rest of the new code in this function came from, and why it's needed :)

It's needed for https://github.com/JetBrains/kotlin/blob/45b1c33ed8a33c92207d1bd9f2e83738995836ae/compiler/cli/cli-base/src/org/jetbrains/kotlin/cli/jvm/compiler/KotlinCoreEnvironment.kt#L208
The compiler does the same  [here](https://github.com/JetBrains/kotlin/blob/45b1c33ed8a33c92207d1bd9f2e83738995836ae/compiler/cli/src/org/jetbrains/kotlin/cli/jvm/jvmArguments.kt#L149)

> As far as I can see, the called compiler function [already checks](https://github.com/JetBrains/kotlin/blob/1.9.0/compiler/cli/cli-base/src/org/jetbrains/kotlin/cli/jvm/config/JvmContentRoots.kt#L94) for java.home

It does not put `JVMConfigurationKeys.JDK_HOME` in the configuration.

**Pull request merged**